File: src/charting/data/ScatterData.java

package com.github.mikephil.charting.data;

import com.github.mikephil.charting.interfaces.datasets.IScatterDataSet;

import java.util.List;

public class ScatterData extends BarLineScatterCandleBubbleData<IScatterDataSet> {

    public ScatterData() {
        super();
    }

    public ScatterData(List<IScatterDataSet> dataSets) {
        super(dataSets);
    }

    public ScatterData(IScatterDataSet... dataSets) {
        super(dataSets);
    }

    /**
     * Returns the maximum shape-size across all DataSets.
     *
     * @return
     */
    public getGreatestShapeSize() {

        let max = 0;

        for (IScatterDataSet set : this.mDataSets) {
            let size = set.getScatterShapeSize();

            if (size > max)
                max = size;
        }

        return max;
    }
}

File: src/charting/data/ScatterData.ts
import { BarLineScatterCandleBubbleData } from './BarLineScatterCandleBubbleData';
import { ScatterDataSet } from './ScatterDataSet';
import { Entry } from './Entry';

export class ScatterData extends BarLineScatterCandleBubbleData<Entry, ScatterDataSet> {
    /**
     * Returns the maximum shape-size across all DataSets.
     */
    public get greatestShapeSize() {
        let max = 0;
        for (let index = 0; index < this.mDataSets.length; index++) {
            const size = this.mDataSets[index].scatterShapeSize;

            if (size > max) {
                max = size;
            }
        }
        return max;
    }
}
